pkg/components/internal/testutil/jsonpath.go

```diff
@@ -38,6 +43,60 @@ func unstructredObj(t *testing.T, yamlObj string) *unstructured.Unstructured {
 	return u
 }
 
+// removeYAMLComments converts YAML to JSON and back again, this removes the comments in the YAML
+// and any extra whitespaces spaces.
+func removeYAMLComments(t *testing.T, yamlObj []byte) []byte {
```

Comment:
I think both those functions should return error and then we can wrap them with function utilizing `*testing.T`, so we keep the convention of passing errors up. It makes code easier to read and test.

Reply:
Done.